src/renderer/common/hooks/useKeyboardShortcut.ts

```diff
@@ -0,0 +1,23 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import * as React from "react";
+import { TKeyboardShortcutReadOnly } from "readium-desktop/common/keyboard";
+import { registerKeyboardListener, unregisterKeyboardListener } from "../keyboard";
+import { useSelector } from "./useSelector";
+import { ICommonRootState } from "readium-desktop/common/redux/states/commonRootState";
+
+export function useKeyboardShortcut(ListenForKeyUP: boolean, keyboardShortcut: (s: ICommonRootState["keyboard"]["shortcuts"]) => TKeyboardShortcutReadOnly, callback: () => void) {
+
+    const keyboardShortcutState = useSelector((state: ICommonRootState) => state.keyboard.shortcuts);
+    React.useEffect(() => {
+        registerKeyboardListener(ListenForKeyUP, keyboardShortcut(keyboardShortcutState), callback);
+        return () => unregisterKeyboardListener(callback);
+    }, [keyboardShortcutState]);
```

Comment:
I haven't tested this hook placeholder yet, but I think this will cause unnecessary renders due to the "state" object inside the dependency array being compared via reference identity. See `keyboardShortcutsMatch()` calls in `componentDidUpdate()` class component lifecycle.
I also don't see the `ensureKeyboardListenerIsInstalled()` call equivalent to the `componentDidMount()` class component lifecycle (`React.useEffect()` with _empty dependency array_)
@panaC did you test this hook, or is this work in progress? (the PR is marked as "[DEV]")
Thank you :)